tests/lib/e2e/buckets/files.e2e-spec.ts

```diff
@@ -0,0 +1,400 @@
+          .post('/buckets')
+          .set('Authorization', getAuth(testUser))
+
+        // Act: remove the file
+        const fakeFileId = new ObjectId()
+        const response = await testServer.delete(`/buckets/${bucketId}/files/${fakeFileId}`)
+          .set('Authorization', getAuth(testUser))
+
+        // Assert
+        expect(response.status).toBe(404);
+
+      })
+    })
+
+    describe('Sharing a file', () => {
+      it('When a user wants to share a file, it should be able to create a token for the bucket passing PULL as operation', async () => {
+
+        // Arrange: Create a bucket
+        const { body: { id: bucketId } } = await testServer
+          .post('/buckets')
+          .set('Authorization', getAuth(testUser))
+
+        // Act
+        const response = await testServer.post(`/buckets/${bucketId}/tokens`)
+          .set('Authorization', getAuth(testUser))
+          .send({ operation: 'PULL', })
+
+        // Assert
+        expect(response.status).toBe(201);
+
+        const { body } = response
+
+        expect(body.bucket).toBeDefined()
+        expect(body.operation).toBeDefined()
+        expect(body.expires).toBeDefined()
+        expect(body.token).toBeDefined()
+        expect(body.encryptionKey).toBeDefined()
+        expect(body.id).toBeDefined()
+      })
+      it('When a user wants to share a file, it should be able to create a token for the bucket passing PULL as operation and an existing file', async () => {
+
+        // Arrange: Create a bucket
+        const { body: { id: bucketId } } = await testServer
+          .post('/buckets')
+          .set('Authorization', getAuth(testUser))
+
+        // Arrange: start the upload
+        const { body: { uploads: [upload] } } = await testServer.post(`/v2/buckets/${bucketId}/files/start`)
+          .set('Authorization', getAuth(testUser))
+          .send({ uploads: [{ index: 0, size: 1000, }], })
+
+        // Arrange: finish the upload
+        const index = crypto.randomBytes(32).toString('hex');
+        const { body: file } = await testServer.post(`/v2/buckets/${bucketId}/files/finish`)
+          .set('Authorization', getAuth(testUser))
+          .send({
+            index,
+            shards: [{ hash: crypto.randomBytes(20).toString('hex'), uuid: upload.uuid, }],
+          });
+
+        // Act
+        const response = await testServer.post(`/buckets/${bucketId}/tokens`)
+          .set('Authorization', getAuth(testUser))
+          .send({ operation: 'PULL', file: file.id })
+
+        // Assert 
+        expect(response.status).toBe(201);
+
+        const { body } = response
+
+        expect(body.bucket).toBeDefined()
+        expect(body.operation).toBeDefined()
+        expect(body.expires).toBeDefined()
+        expect(body.token).toBeDefined()
+        expect(body.id).toBeDefined()
+        expect(body.encryptionKey).toBeDefined()
+        expect(body.mimetype).toBeDefined()
+        expect(body.size).toBeDefined()
```

Comment:
Same here, let's validate the format of everything plus matching whatever is possible to be matched, in this case, the size, the bucket, the operation, etc.